### PR TITLE
Dry run

### PR DESCRIPTION
**Pull request by iwko**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 setuptools.setup(
     name="biggerquery",
-    version="0.3.0",
+    version="0.4.0dev1",
```
Comment (beczkowb) [inline setup.py]: 0.4.dev1

File: biggerquery/dataset_manager.py [hunk 2]
```diff
@@ -203,8 +209,16 @@ def create_table(self, create_query):
         return job.result()
 
     def collect(self, sql):
-        self.logger.info('COLLECTING DATA: %s', sql)
-        return self.bigquery_client.query(sql).to_dataframe()
+        return self._collect(sql).to_dataframe()
+
+    def estimate_costs(self, sql):
```
Comment (beczkowb) [inline biggerquery/dataset_manager.py]: Add this method to the interactive.py layer.

File: biggerquery_tutorial/part1.ipynb [hunk 3]
```diff
@@ -198,6 +198,27 @@
...
+    "The estimate_costs method returns string with estimated number of bytes read by the query and estimated cost of query in dollars."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "costs = dataset.estimate_costs('''\n",
```
Comment (beczkowb) [inline biggerquery_tutorial/part1.ipynb]: It won't work. You didn't add estimate_costs to interactive layer.

Comment (iwko) [reply biggerquery_tutorial/part1.ipynb]: Done

File: biggerquery/dataset_manager.py [hunk 4]
```diff
@@ -53,6 +53,9 @@ def create_table_id(self, table_name):
     def collect(self, sql, custom_run_datetime=None):
         return self.dataset_manager.collect(sql.format(**self.template_variables(custom_run_datetime)))
 
+    def estimate_costs(self, sql, custom_run_datetime=None):
```
[the remark this answers is not in the document]
Comment (iwko) [reply biggerquery/dataset_manager.py]: Done

File: biggerquery/interactive.py [hunk 5]
```diff
@@ -78,6 +78,14 @@ def collect(self, sql):
             sql,
             operation_name=DEFAULT_OPERATION_NAME)
 
+    def dry_run(self, sql):
```
Comment (beczkowb) [inline biggerquery/interactive.py]: You should unit test those metods

File: biggerquery_tutorial/part1.ipynb [hunk 6]
```diff
@@ -198,6 +198,27 @@
     "all_transactions_from_single_day.peek('2019-01-01')"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Estimating the cost of query\n",
```
Comment (beczkowb) [inline biggerquery_tutorial/part1.ipynb]: Estimating the cost of a query

File: biggerquery_tutorial/part1.ipynb [hunk 7]
```diff
@@ -198,6 +198,27 @@
...
+    "The estimate_costs method returns string with estimated number of bytes read by the query and estimated cost of query in dollars."
```
Comment (beczkowb) [inline biggerquery_tutorial/part1.ipynb]: The estimate_costs method returns a string with the estimated number of bytes read by the query and estimated cost of a query in dollars.

Comment (iwko) [reply biggerquery_tutorial/part1.ipynb]: Done

Comment (iwko) [reply biggerquery_tutorial/part1.ipynb]: Done

File: biggerquery/interactive.py (hunk 5 above)
Comment (iwko) [reply biggerquery/interactive.py]: Done

File: setup.py (hunk 1 above)
Comment (iwko) [reply setup.py]: Done. version=0.4.0

File: biggerquery/dataset_manager.py [hunk 8]
```diff
@@ -224,6 +238,28 @@ def table_exists(self, table_name):
             .to_dataframe()['table_exists'] \
             .iloc[0] > 0
 
+    def _collect(self, sql, job_config=None):
```
[the remark this answers is not in the document]
Comment (iwko) [reply biggerquery/dataset_manager.py]: I've changed method name to `query` but i don't feel that keyword argument will improve anything

File: e2e/test_dataset_manager.py [hunk 9]
```diff
@@ -428,6 +428,22 @@ def test_should_resolve_custom_partition(self):
 
 class CollectTestCase(DatasetManagerTestCase):
 
+    def test_dry_run(self):
```
[the remark this answers is not in the document]
Comment (iwko) [reply e2e/test_dataset_manager.py]: Done

**Pull request closed without merging**